lib/solr.ts

```diff
@@ -106,12 +106,14 @@ export class Client {
       ? 'https://'
       : 'http://';
 
-    this.undiciClient = new UndiciClient(
-      `${urlPrefix}${this.options.host}:${this.options.port}`,
-      {
-        connect: this.options.tls,
-      }
-    );
+	  const port = options.noPort ? '' : `:${this.options.port}`;
+
+	  this.undiciClient = new UndiciClient(
```

Comment:
it would be nice to have test for this. I think undici has built-in mocks, you can use that to avoid sending real request

Reply:
I agree, but I'm not sure how we'd be able to get the docker image to listen without specifying the port. Do you know how that could work?